### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by Athou** [Athou/lapinstance#5]
When the backend returns a specific error and not a generic one, display a better message in the client.

**Comment by kedbirhan** [comment]
Hi @Athou I was integrating spring backend to a react front end. i am serving the react frontend from the spring backend like lapInstance. but when a user refreshes a page,  404 error is shown instead of reloading the page. is there a way I can prevent that?.  I searched your backend implementation but I could not find any configurations. 
Thanks in Advance

**Comment by Athou** [comment]
Very difficult without seeing your code. Are you using react-router? If so, you probably need to use a hash router like this https://github.com/Athou/lapinstance/blob/master/client/src/App.tsx#L3

**Comment by kedbirhan** [comment]
@Athou  yes, I am using react-router. I did change the router to hash-router as you suggested and it worked without showing the dreaded 404 error. Thanks!

**Comment by kedbirhan** [comment]
HI @Athou  I had an idea about using straight BrowserRouter instead of HashRouter that works on refresh.
My idea is to prepend the UI URLs shown in the browser with "ui" or "app" so we can differentiate it from the restful endpoint. This allows us to refresh the page without getting a Page Not Found error when the user manually refreshes the page. In addition, we will have to add in the backend to route any url that starts with "app" or "ui" to index.html as shown in the code below.

the reason is having hash routing doesn't look clean and its also not [recommended](https://reactrouter.com/en/main/router-components/hash-router). I have tried it in this project and it works fine. let me know if you want me to make a pull request.

``` java
@Configuration
public class WebConfiguration implements WebMvcConfigurer {

	@Override
	public void addViewControllers(ViewControllerRegistry registry) {
		registry.addViewController("/app/**").setViewName("forward:/index.html");
	}
}

```
